Lib/test/test_typing.py

```diff
@@ -3809,6 +3809,23 @@ class Y(C[int]):
         self.assertEqual(Y.__qualname__,
                          'GenericTests.test_repr_2.<locals>.Y')
 
+    def test_repr_3(self):
+        T = TypeVar('T')
+        P = ParamSpec('P')
+
+        class MyCallable(Generic[P, T]):
+            pass
+
+        object_to_expected_repr = {
+            MyCallable[[], bool]:         "MyCallable[[], bool]",
+            MyCallable[[int], bool]:      "MyCallable[[int], bool]",
+            MyCallable[[int, str], bool]: "MyCallable[[int, str], bool]"
+        }
+
+        for obj, expected_repr in object_to_expected_repr.items():
+            with self.subTest(obj=obj, expected_repr=expected_repr):
+                self.assertEqual(repr(obj), f"{MyCallable.__module__}.{expected_repr}")
```

Comment:
Sorry for the noise. I just tried it out locally, and it seems that this still won't pass, as the repr of the first one is in fact `test.test_typing.GenericTests.test_repr_3.<locals>.MyCallable[[], bool]` rather than `test.test_typing.MyCallable[[], bool]`

```suggestion
                self.assertRegex(
                    repr(obj),
                    fr"^{re.escape(MyCallable.__module__)}.*\.{re.escape(expected_repr)}$"
                )
```